src/System.Drawing.Common/src/System/Drawing/Printing/MarginsConverter.cs

```diff
@@ -0,0 +1,117 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.ComponentModel;
+using System.ComponentModel.Design.Serialization;
+using System.Globalization;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace System.Drawing.Printing
+{
+	/// <summary>
+	/// Summary description for MarginsConverter.
+	/// </summary>
+	public class MarginsConverter : ExpandableObjectConverter
+    {
+		public MarginsConverter() { }
+
+		public override bool CanConvertFrom(ITypeDescriptorContext context, Type sourceType)
+        {
+			if (sourceType == typeof(string))
+				return true;
+			
+			return base.CanConvertFrom(context, sourceType);
+		}
+
+		public override bool CanConvertTo(ITypeDescriptorContext context, Type destinationType)
+        {
+			if (destinationType == typeof(string))
+				return true;
+			
+			if (destinationType == typeof (InstanceDescriptor))
+				return true;
+
+			return base.CanConvertTo(context, destinationType);
+		}
+
+		public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo culture, object value)
+        {
+			if (value is string)
+			{
+				if (value == null)
+					return new Margins();
+				
+				// format [left];[right];[top];[bottom]
+				string separator = @"( |\t)*";
+				separator = separator + ";" + separator;
+				string regex = @"(?<left>\d+)" + separator + @"(?<right>\d+)" + separator + @"(?<top>\d+)" + separator + @"(?<bottom>\d+)";
+				
+				Match match = new Regex(regex).Match(value as string);
```

Comment:
Nit: We can avoid this cast by changing the initial `if (value is string)` earlier to be `if (value is string valueString)` and then just use `valueString` here.